From dcea7a7cceb7378b659553e52b0e4ca3baf44f7b Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <chrisg@marvell.com>
Date: Mon, 25 Jun 2018 20:44:33 +0300
Subject: [PATCH 0781/1200] ap807: sar: Enable sampled-at-reset driver for
 AP807

- This patch enables the sampled-at-reset driver for AP807
  by adding the driver's compatible to AP807 SoC level DT

Change-Id: I5168c8bee1d3e3c94d5ec2e989caaa487780dfca
Signed-off-by: Christine Gharzuzi <chrisg@marvell.com>
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/57318
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/57472
---
 arch/arm/dts/armada-ap806.dtsi  | 9 +++++++++
 arch/arm/dts/armada-ap807.dtsi  | 8 ++++++++
 arch/arm/dts/armada-ap80x.dtsi  | 9 ---------
 drivers/misc/mvebu_sar/Makefile | 1 +
 4 files changed, 18 insertions(+), 9 deletions(-)

diff --git a/arch/arm/dts/armada-ap806.dtsi b/arch/arm/dts/armada-ap806.dtsi
index ec86fdc24f..6a8eced524 100644
--- a/arch/arm/dts/armada-ap806.dtsi
+++ b/arch/arm/dts/armada-ap806.dtsi
@@ -54,6 +54,15 @@
 
 	AP_NAME {
 		config-space {
+			sar-reg {
+				compatible = "marvell,sample-at-reset-common",
+					     "marvell,sample-at-reset-ap806";
+				reg = <0x6F8200 0x8>;
+				sar-driver = "ap806_sar";
+				sar-name = "ap806_sar";
+				status = "okay";
+			};
+
 			thermal: thermal@6f8084 {
 				compatible = "marvell,mvebu-thermal", "marvell,thermal-ext-sensor";
 				reg = <0x6f8084 0x12>;
diff --git a/arch/arm/dts/armada-ap807.dtsi b/arch/arm/dts/armada-ap807.dtsi
index e68795e9e9..5bd9f9939c 100644
--- a/arch/arm/dts/armada-ap807.dtsi
+++ b/arch/arm/dts/armada-ap807.dtsi
@@ -19,6 +19,14 @@
 
 	AP_NAME {
 		config-space {
+			sar-reg {
+				compatible = "marvell,sample-at-reset-common",
+					     "marvell,sample-at-reset-ap807";
+				reg = <0x6F8200 0x8>;
+				sar-driver = "ap807_sar";
+				sar-name = "ap807_sar";
+				status = "okay";
+			};
 
 			thermal: thermal@6f8084 {
 				compatible = "marvell,mvebu-thermal",
diff --git a/arch/arm/dts/armada-ap80x.dtsi b/arch/arm/dts/armada-ap80x.dtsi
index 4e6d3aa86a..cf9f2ca5c2 100644
--- a/arch/arm/dts/armada-ap80x.dtsi
+++ b/arch/arm/dts/armada-ap80x.dtsi
@@ -63,15 +63,6 @@
 				#gpio-cells = <2>;
 			};
 
-			sar-reg {
-				compatible = "marvell,sample-at-reset-common",
-					     "marvell,sample-at-reset-ap806";
-				reg = <0x6F8200 0x8>;
-				sar-driver = "ap806_sar";
-				sar-name = "ap806_sar";
-				status = "okay";
-			};
-
 			ap_spi0: spi@510600 {
 				compatible = "marvell,armada-380-spi";
 				reg = <0x510600 0x50>;
diff --git a/drivers/misc/mvebu_sar/Makefile b/drivers/misc/mvebu_sar/Makefile
index d0f73dd981..03086b88d8 100644
--- a/drivers/misc/mvebu_sar/Makefile
+++ b/drivers/misc/mvebu_sar/Makefile
@@ -1,4 +1,5 @@
 
 obj-$(CONFIG_MVEBU_SAR) += sar-uclass.o
 obj-$(CONFIG_TARGET_MVEBU_ARMADA_8K) += ap806_sar.o
+obj-$(CONFIG_TARGET_MVEBU_ARMADA_8K) += ap807_sar.o
 obj-$(CONFIG_TARGET_MVEBU_ARMADA_8K) += cp110_sar.o
-- 
2.22.0

